packages/ncdu2/fix-progname.patch

```diff
@@ -0,0 +1,214 @@
+@@ -44,7 +44,7 @@ my $maxitemsperblock = 0;
+ 
+ {
+     die "Input too short\n" if 8 != read STDIN, my $sig, 8;
+-    die "Invalid file signature\n" if $sig ne "\xbfncduEX1";
++    die "Invalid file signature\n" if $sig ne "\xbfncdu2EX1";
+ }
+ 
+ my @itemkeys = qw/
+diff --git a/src/bin_export.zig b/src/bin_export.zig
+index 08be2c6..88891e1 100644
+--- a/src/bin_export.zig
++++ b/src/bin_export.zig
+@@ -21,7 +21,7 @@ pub const global = struct {
+ 
+ const BLOCK_SIZE: usize = 64*1024;
+ 
+-pub const SIGNATURE = "\xbfncduEX1";
++pub const SIGNATURE = "\xbfncdu2EX1";
+ 
+ pub const ItemKey = enum(u5) {
+     // all items
+diff --git a/src/main.zig b/src/main.zig
+index bdf98ef..41d68c4 100644
+--- a/src/main.zig
++++ b/src/main.zig
+@@ -38,7 +38,7 @@ test "imports" {
+ }
+ 
+ // "Custom" allocator that wraps the libc allocator and calls ui.oom() on error.
+-// This allocator never returns an error, it either succeeds or causes ncdu to quit.
++// This allocator never returns an error, it either succeeds or causes ncdu2 to quit.
+ // (Which means you'll find a lot of "catch unreachable" sprinkled through the code,
+ // they look scarier than they are)
+ fn wrapAlloc(_: *anyopaque, len: usize, ptr_alignment: u8, return_address: usize) ?[*]u8 {
+@@ -341,14 +341,14 @@ fn tryReadArgsFile(path: [:0]const u8) void {
+ 
+ fn version() noreturn {
+     const stdout = std.io.getStdOut();
+-    stdout.writeAll("ncdu " ++ program_version ++ "\n") catch {};
++    stdout.writeAll("ncdu2 " ++ program_version ++ "\n") catch {};
+     std.process.exit(0);
+ }
+ 
+ fn help() noreturn {
+     const stdout = std.io.getStdOut();
+     stdout.writeAll(
+-    \\ncdu <options> <directory>
++    \\ncdu2 <options> <directory>
+     \\
+     \\Options:
+     \\  -h,--help                  This help message
+@@ -367,11 +367,11 @@ fn help() noreturn {
+     \\  -L, --follow-symlinks      Follow symbolic links (excluding directories)
+     \\  --exclude-caches           Exclude directories containing CACHEDIR.TAG
+     \\  --exclude-kernfs           Exclude Linux pseudo filesystems (procfs,sysfs,cgroup,...)
+-    \\  --confirm-quit             Confirm quitting ncdu
++    \\  --confirm-quit             Confirm quitting ncdu2
+     \\  --color SCHEME             Set color scheme (off/dark/dark-bg)
+     \\  --ignore-config            Don't load config files
+     \\
+-    \\Refer to `man ncdu` for the full list of options.
++    \\Refer to `man ncdu2` for the full list of options.
+     \\
+     ) catch {};
+     std.process.exit(0);
+@@ -492,14 +492,14 @@ pub fn main() void {
+     };
+ 
+     if (loadConf) {
+-        tryReadArgsFile("/etc/ncdu.conf");
++        tryReadArgsFile("@TERMUX_PREFIX@/etc/ncdu2.conf");
+ 
+         if (std.posix.getenvZ("XDG_CONFIG_HOME")) |p| {
+-            const path = std.fs.path.joinZ(allocator, &.{p, "ncdu", "config"}) catch unreachable;
++            const path = std.fs.path.joinZ(allocator, &.{p, "ncdu2", "config"}) catch unreachable;
```

Comment:
> [!IMPORTANT]
> This change does modify the config location for `ncdu2` which may be undesirable.

I have taken care to also modify the corresponding section of the man page,
but changing the config location from `XDG_CONFIG_HOME/ncdu/config` to `XDG_CONFIG_HOME/ncdu/config` may still cause undue confusion to users.

The alternative would be sharing the config location between the two `ncdu`'s,
which may lead to incompatibilities.
<sub>Although who's realistically gonna use both?</sub>

Reply:
I wonder if it's worth the trouble and potential confusion to do the `ncdu2` rename at all, and if we instead can let this package conflict with `TERMUX_PKG_CONFLICTS="ncdu"`, so only one can be installed at the same time?

Reply:
Is there even a feature difference between `ncdu` and `ncdu2` at this point?<sub>(rhet. question)</sub>
We may as well just drop `ncdu` 1.x and replace it with 2.x if there is not.

Reply:
The zig based packages often have a minimum kernel requirement https://github.com/termux/termux-packages/issues/20783#issuecomment-2211719239. I personally havent verified ncdu2 on old kernel. I expect C based 1.x can keep working for really old devices.

Reply:
How old are we talking here?
Is there any official documentation from Zig on the minimum supported Linux kernel version?

Reply:
I think deliberating whether to keep both `ncdu` packages around can be left for another time.
For now, if we're keeping both around I think they should have separate config files, and the `--help` output and man page for `ncdu2` should mention `ncdu2` as the program name.